projects/extension/sql/idempotent/905-text-to-sql.sql

```diff
@@ -0,0 +1,510 @@
+, logit_bias pg_catalog.jsonb default null
+, logprobs pg_catalog.bool default null
+, top_logprobs pg_catalog.int4 default null
+, max_tokens pg_catalog.int4 default null
+, n pg_catalog.int4 default null
+, presence_penalty pg_catalog.float8 default null
+, seed pg_catalog.int4 default null
+, stop pg_catalog.text default null
+, temperature pg_catalog.float8 default null
+, top_p pg_catalog.float8 default null
+, openai_user pg_catalog.text default null
+) returns pg_catalog.jsonb
+as $func$
+    select json_object
+    ( 'provider': 'openai'
+    , 'model': model
+    , 'api_key': api_key
+    , 'api_key_name': api_key_name
+    , 'base_url': base_url
+    , 'frequency_penalty': frequency_penalty
+    , 'logit_bias': logit_bias
+    , 'logprobs': logprobs
+    , 'top_logprobs': top_logprobs
+    , 'max_tokens': max_tokens
+    , 'n': n
+    , 'presence_penalty': presence_penalty
+    , 'seed': seed
+    , 'stop': stop
+    , 'temperature': temperature
+    , 'top_p': top_p
+    , 'openai_user': openai_user
+    absent on null
+    )
+$func$ language sql immutable security invoker
+set search_path to pg_catalog, pg_temp
+;
+
+-------------------------------------------------------------------------------
+-- text_to_sql_ollama
+create or replace function ai.text_to_sql_ollama
+( model pg_catalog.text
+, host pg_catalog.text default null
+, keep_alive pg_catalog.text default null
+, chat_options pg_catalog.jsonb default null
+) returns pg_catalog.jsonb
+as $func$
+    select json_object
+    ( 'provider': 'ollama'
+    , 'model': model
+    , 'host': host
+    , 'keep_alive': keep_alive
+    , 'chat_options': chat_options
+    absent on null
+    )
+$func$ language sql immutable security invoker
+set search_path to pg_catalog, pg_temp
+;
+
+-------------------------------------------------------------------------------
+-- text_to_sql_anthropic
+create or replace function ai.text_to_sql_anthropic
+( model text
+, max_tokens int default 1024
+, api_key text default null
+, api_key_name text default null
+, base_url text default null
+, timeout float8 default null
+, max_retries int default null
+, user_id text default null
+, stop_sequences text[] default null
+, temperature float8 default null
+, top_k int default null
+, top_p float8 default null
+) returns pg_catalog.jsonb
+as $func$
+    select json_object
+    ( 'provider': 'anthropic'
+    , 'model': model
+    , 'max_tokens': max_tokens
+    , 'api_key': api_key
+    , 'api_key_name': api_key_name
+    , 'base_url': base_url
+    , 'timeout': timeout
+    , 'max_retries': max_retries
+    , 'user_id': user_id
+    , 'stop_sequences': stop_sequences
+    , 'temperature': temperature
+    , 'top_k': top_k
+    , 'top_p': top_p
+    absent on null
+    )
+$func$ language sql immutable security invoker
+set search_path to pg_catalog, pg_temp
+;
+
+-------------------------------------------------------------------------------
+-- text_to_sql
+create or replace function ai.text_to_sql
```

Comment:
ugh this seem so repetitive and redundant to functions like ai.openai_chat_complete. Maybe we just have function to generate the full prompt and have people call the chat_complete functions explicitly?